### PR TITLE
feat(theme-default): support heroImageDark frontmatter (close #526)

### LINKED ISSUE #526: [Feature request] heroImage change with dark mode
Just like logo & logoDark in themeConfig, we should also support heroImageDark in base config or home formatter

### PR DESCRIPTION
**Pull request by qcyblm**
close #526

### CODE REVIEW

File: packages/@vuepress/theme-default/src/client/components/Home.vue [hunk 1]
```diff
@@ -50,18 +50,19 @@ import {
...
 const frontmatter = usePageFrontmatter<DefaultThemeHomePageFrontmatter>()
 const siteLocale = useSiteLocaleData()
+const isDarkMode = useDarkMode()
 
 // hero image and title
 const heroImage = computed(() => {
-  if (!frontmatter.value.heroImage) {
-    return null
+  if (isDarkMode.value && frontmatter.value.heroImageDark !== undefined) {
+    return frontmatter.value.heroImageDark
```
Comment (alirezanet) [inline packages/@vuepress/theme-default/src/client/components/Home.vue:63]: I think you should use withBase, 
``` ts
const heroImage = computed(() => {
  if (frontmatter.value.heroImageDark && isDarkMode.value) {
    return withBase(frontmatter.value.heroImageDark)
  }
  if (frontmatter.value.heroImage) {
    return withBase(frontmatter.value.heroImage)
  }
  return null
})
```

Comment (qcyblm) [reply packages/@vuepress/theme-default/src/client/components/Home.vue:63]:
```
<img v-if="heroImage" :src="withBase(heroImage)" :alt="heroAlt" />
```

**Comment by alirezanet** [comment]
any update on this? :(
you can find a real-world example [here](https://alirezanet.github.io/Gridify/) why we need this

File: packages/@vuepress/theme-default/src/client/components/Home.vue [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <template>
   <main class="home" :aria-labelledby="heroText ? 'main-title' : undefined">
     <header class="hero">
-      <img v-if="heroImage" :src="heroImage" :alt="heroAlt" />
+      <img v-if="heroImage" :src="withBase(heroImage)" :alt="heroAlt" />
```
Comment (meteorlxy) [inline packages/@vuepress/theme-default/src/client/components/Home.vue:4]: Also need to wrap the image with `<ClientOnly>`

**Review by meteorlxy — CHANGES_REQUESTED** [review]

**Review by meteorlxy — APPROVED** [review]
LGTM

**Pull request merged**